### ISSUE TITLE
Streaming radio audio url never calls the callback of request

### ISSUE THREAD

**Issue by dinostheo** [request/request#2656]
An audio url from a radio station hit my application, which is using request to check the incoming urls. The application just hangs on the specific url `http://streaming.radionomy.com/RadioGansta`

I tried the following with this url and nothing was printed.

```js
const request = require('request');

request('http://streaming.radionomy.com/RadioGansta', (error, response, body) => {
	console.log('Error: ', error);
	console.log('Response: ', response);
	console.log('Body: ', body);
});
```

I run the following with the debug flag enabled and I got the following
```
REQUEST { followRedirect: false,
  timeout: 15000,
  gzip: true,
  headers: { 'user-agent': 'Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)' },
  uri: 'http://streaming.radionomy.com/RadioGansta',
  callback: [Function] }
REQUEST make request http://streaming.radionomy.com/RadioGansta
REQUEST onRequestResponse http://streaming.radionomy.com/RadioGansta 200 { 'content-type': 'audio/mpeg',
  date: 'Sun, 07 May 2017 09:31:21 GMT',
  'icy-br': '128, 128',
  'ice-audio-info': 'bitrate=128;samplerate=44100;channels=2',
  'icy-description': 'RadioGansta la radio des artistes ind&#xe9;pendants et surtout RadioGansta  c est votre radio. Le meilleur de la musique urban rap rnb reggae ragga  ',
  'icy-genre': 'rap',
  'icy-name': 'RadioGansta',
  'icy-pub': '1',
  'icy-url': 'http://www.radiogansta.fr',
  server: 'Icecast 2.3.3-kh8',
  'cache-control': 'no-cache, no-store',
  pragma: 'no-cache',
  'access-control-allow-origin': '*',
  'access-control-allow-headers': 'Origin, Accept, X-Requested-With, Content-Type',
  'access-control-allow-methods': 'GET, OPTIONS, HEAD',
  connection: 'close',
  expires: 'Mon, 26 Jul 1997 05:00:00 GMT' }
REQUEST reading response's body
REQUEST finish init function http://streaming.radionomy.com/RadioGansta
```

It feels to me like a bug, as I would expect at least to get a call to my callback function and then handle the response or even to throw an exception. 
Am I dealing with this in the wrong way?

**Comment by dinostheo** [comment]
I switched to stream in order to detect the type of response and on audio or video I abort the request. However, shouldn't I have got a timeout, as it's set?

**Comment by bradisbell** [comment]
This isn't a bug.  You're retrieving a radio stream, which has indefinite length.  Since you're passing a callback when you make the request, the module is going to buffer the data until you run out of memory.

You should handle `.on('response', ...)` instead, as indicated in the guide on streaming.  https://github.com/request/request#streaming

**Comment by dinostheo** [comment]
@bradisbell That's exactly what I did 😃
The problem is that I don't know the urls beforehand so I had to check the headers to detect unsupported content-types. My biggest issue here is that I was not getting a timeout or perhaps having an option that if the response is not complete in N seconds to abort the request. Anyways, fetching the headers first helps in identifying those urls. 

I was expecting for someone to close this issue, but I also think that is not an issue anymore, so I will close it myself.
